services/QuillLMS/app/controllers/sales_form_submission_controller.rb

```diff
@@ -27,10 +27,20 @@ def create
 
   def options_for_sales_form
     type = params[:type]
-    schools_or_districts = type.classify.constantize.all.pluck(:name)
+    search_query = params[:search]
+    prefix = search_query_prefix(search_query)
+    schools_or_districts = type.classify.constantize.all.where("lower(name) LIKE :prefix", prefix: "#{prefix.downcase}%").limit(10).pluck(:name)
     render json: { options: schools_or_districts }
   end
 
```

Comment:
For the regex in this method, `/\d{5}/`, in some cases might not give us quite what we want:

```
pry(main)> search = "123 The 23456 GoodZ 7890124"                                            
=> "123 The 23456 GoodZ 7890124"
pry(main)> search.gsub(/\d{5}/,'').strip                                                           
=> "123 The  GoodZ 24"
```

`/d*/` might be more appropriate here: 
```
pry(main)> search.gsub(/\d*/,'').strip                                                             
=> "The  GoodZ"
```

This leaves double spaces in the middle but you can use [squish](https://duetngyen.wordpress.com/2016/07/20/hello/)
```
pry(main)> search.gsub(/\d*/,'').squish                                                            
=> "The GoodZ"
```

Reply:
great! this is much better, thanks for the examples